### PR TITLE
Code Optimization 

### PR DESCRIPTION
**Pull request by fqiang**
This PR provides a MPIWrapper type that handles the MPI initialization and communicator when user doesn't provide one. The PR also was tested to work with optimized StructJuMPSolverInterface code.

### CODE REVIEW

File: src/StructJuMP.jl [hunk 1]
```diff
@@ -13,15 +13,33 @@ export StructuredModel, getStructure, getparent, getchildren, getProcIdxSet,
 # ---------------
 # StructureData
 # ---------------
+type MPIWrapper
+    comm::MPI.Comm
+    userInitMPI::Bool
+
+    function MPIWrapper(comm, user)
+        @show comm, user
```
Comment (joehuchette) [inline src/StructJuMP.jl]: leftover debugging code

File: src/StructJuMP.jl [hunk 2]
```diff
@@ -34,27 +52,27 @@ default_probability(::Void) = 1.0
...
-    if parent !== nothing
-        @assert id != 0 
-        stoch = getStructure(parent)
-        stoch.children[id] = m
-        push!(stoch.probability, prob)
-        comm = stoch.comm
-        userInitMPI = stoch.userInitMPI
-        @assert comm == MPI.COMM_WORLD
-    else
+    if parent == nothing
```
Comment (joehuchette) [inline src/StructJuMP.jl]: little thing, but can be ``parent === nothing``

**Comment by joehuchette** [comment]
LGTM, besides the debugging code.

File: src/StructJuMP.jl (hunk 2 above)
Comment (fqiang) [inline src/StructJuMP.jl]: Hi Joey, do you know the difference between "===" and "==" ? @joehuchette

Comment (joehuchette) [inline src/StructJuMP.jl]: ``==`` is "programatic" equality, ``===`` is object identity (see the help output for ``==`` and ``===`` for more info). ``===`` will in general be more efficient, so this was just a micro optimization note for next time.

**Comment by joehuchette** [comment]
Bump on the debugging stuff @fqiang

File: src/StructJuMP.jl (hunk 2 above)
Comment (fqiang) [inline src/StructJuMP.jl]: It is nice to know. Thanks @joehuchette

**Comment by fqiang** [comment]
@joehuchette The debugging and "===" micro optimization are both applied. Thanks for reviewing the code.

**Pull request merged**